### ISSUE TITLE
Game code severely out of date, shader code broken, mods require six year old 'upcoming' binaries.

### ISSUE THREAD

**Issue by RoyaleNoir** [ValveSoftware/source-sdk-2013#513]
This repository's code was last updated in 2015, and since then there have been multiple notable updates to Half-Life 2, such as the infamous [blinking fix](https://www.theverge.com/tldr/2019/9/27/20887059/half-life-2-blinking-bug-fixed-valve-steampipe), which are yet to be included in this codebase.

Even with the game code being out of date, it still requires the base to be set to the "upcoming" branch for the mod to work, which can be counter-intuitive to the average player, especially given that the binaries on that branch are well over six years old at this point.

The shader code is in an even worse state with it just kind of being placed into the repository without checking for functionality, as major changes to almost every shader's cpp files need to be made for them to compile, and every shader has to be renamed by hand to avoid overriding which causes and engine error.

**Comment by Adrianilloo** [comment]
The trees wave their leaves, the birds don't fear, no enemies appear. In the middle of the horizon, the silence is clear.

**Comment by SC1040-TS2** [comment]
> Another request I would have is to finally push the 'upcoming' branch's binaries into the 'public' branch on Steam so that players can start mods without having to manually select the branch.

For some amount of context in the event @kisak-valve (Apologies for the ping in advance) ever comes back around here, this quoted line refers to the Mapbase Source project found here: https://github.com/mapbase-source/source-sdk-2013 .

Singleplayer games built off of Mapbase Source will run off of whatever is the latest build version of Source SDK 2013 Singleplayer that is exposed to developers. In this case, it is the severely out of date Upcoming Beta Build version 2422905, compared to the latest Singleplayer versions of HL2, Episode One, and Episode Two all using Build version 5377866.

Pushing the Upcoming Beta Build version 2422905 to Stable would resolve a small amount of headache when trying to run mods using Mapbase, such as Entropy: Zero. However, it would be even more ideal if the Build version 5377866 code was pushed to the Singleplayer SDK instead, and also moved to here as well.

**Comment by RoyaleNoir** [comment]
> 
> 
> > Another request I would have is to finally push the 'upcoming' branch's binaries into the 'public' branch on Steam so that players can start mods without having to manually select the branch.
> 
> For some amount of context in the event @kisak-valve (Apologies for the ping in advance) ever comes back around here, this quoted line refers to the Mapbase Source project found here: https://github.com/mapbase-source/source-sdk-2013 .
> 
> Singleplayer games built off of Mapbase Source will run off of whatever is the latest build version of Source SDK 2013 Singleplayer that is exposed to developers. In this case, it is the severely out of date Upcoming Beta Build version 2422905, compared to the latest Singleplayer versions of HL2, Episode One, and Episode Two all using Build version 5377866.
> 
> Pushing the Upcoming Beta Build version 2422905 to Stable would resolve a small amount of headache when trying to run mods using Mapbase, such as Entropy: Zero. However, it would be even more ideal if the Build version 5377866 code was pushed to the Singleplayer SDK instead, and also moved to here as well.

The issue isn't specific to Mapbase, it also applies to the stock code on this repo.

**Comment by SC1040-TS2** [comment]
I understand that. However, Mapbase Source projects are a use case where this is most obvious.

**Comment by Blixibon** [comment]
> I understand that. However, Mapbase Source projects are a use case where this is most obvious.

This issue applies to the majority of Source 2013 mods today and has existed since long before Mapbase. Mapbase is just another mod which falls under that umbrella and has no reason to be associated with this in any way.

The regular Steam version of Entropy : Zero also does not use Mapbase, although I think it does technically use the Upcoming build.

**Comment by RoyaleNoir** [comment]
yeah i've been messing with source since around 2015-16 and this has *always* been an issue

**Comment by RoyaleNoir** [comment]
Updated the title and description to make it more clear, as #359 is already about a code update, and this is about binaries
